libpod/image/pull.go

```diff
@@ -356,11 +420,36 @@ func checkRemoteImageForLabel(ctx context.Context, label string, imageInfo pullR
 	if err != nil {
 		return err
 	}
+	return isLabelInInspect(label, remoteInspect.Labels, imageInfo.image)
+}
+
+// checkRemoteImageIsUpToDate checks if the remote image has a specific label. If not, an error is returned.
+// Then it checks if an source reference is newer than lastUpdated, or the time an image we are checking was last updated.
+// It returns false if lastUpdate is older than the source image, true otherwise
+func checkRemoteImageIsUpToDate(ctx context.Context, label *string, lastUpdated time.Time, imageInfo pullRefPair, sc *types.SystemContext) (bool, error) {
+	labelImage, err := imageInfo.srcRef.NewImage(ctx, sc)
+	if err != nil {
+		return false, err
+	}
+	remoteInspect, err := labelImage.Inspect(ctx)
+	if err != nil {
+		return false, err
+	}
+	if label != nil {
+		if err = isLabelInInspect(*label, remoteInspect.Labels, imageInfo.image); err != nil {
+			return false, err
+		}
+	}
+	return !remoteInspect.Created.After(lastUpdated), nil
+}
+
+// checks if a label is in the labels map passed in from an image.Inspect
+func isLabelInInspect(label string, labels map[string]string, name string) error {
```

Comment:
Pass all of `remoteInspect` to this function, so that the callers don’t have to both say `.Labels`, and so that the operation of the function actually matches its name.

<s>(Absolutely non-blocking: Consider naming this `imageHasLabel`/`imageContainsLabel`,`labelIsInInspect` or something like that, closer to regular English.)

(In the comment, “checks” is vague; “returns true“ would be more definite.)</s>

… actually, I have almost missed that this _fails_ if the label is missing. I expect a function named with a predicate like “is” to return a bool.

So, name this maybe something like `ensureImageHasLabel` to be clear about the purpose and usage, please. (I realize that `checkRemoteImageForLabel` is not that explicit either… let’s not make this worse.)